etna/datasets/tsdataset.py

```diff
@@ -1123,3 +1159,29 @@ def info(self, segments: Optional[Sequence[str]] = None) -> None:
         # print the results
         result_string = "\n".join(lines)
         print(result_string)
+
+    def to_torch_dataset(
+        self, make_samples: Callable[[pd.DataFrame], Union[Iterator[dict], Iterable[dict]]], dropna: bool = True
```

Comment:
Signature of `make_samples` looks strange. As I remember `Iterator` is `Iterable` because in general its method `iter` returns `self`. So if we need `Iterable` mentioning `Iterator` probably isn't necessary.